content/blog/jamstack-with-pulumi/index.md

```diff
@@ -0,0 +1,359 @@
+
+## Building the Cloud Infrastructure
+
+Configuring a static website can be complicated when using either the AWS web interfaces or the AWS CLI. To deploy a static website, we need to create and configure the following cloud resources.
+
+- [Amazon S3](https://aws.amazon.com/s3/) to store the website's contents
+- [Amazon CloudFront](https://aws.amazon.com/cloudfront/) for the CDN serving content
+- [Amazon Route53](https://aws.amazon.com/route53/) to set up the DNS for the website
+- [Amazon Certificate Manager](https://aws.amazon.com/certificate-manager/) for securing the site via HTTPS
+
+Using Pulumi to deploy Infrastructure as Code, we can create and build our infrastructure using Python or your favorite programming language. If you haven't installed Pulumi and configured it to work with your AWS credentials, follow the [Getting Started with AWS guide]({{< relref "/docs/get-started/aws">}}).
+
+To get started building our infrastructure, we’ll download the Python example for setting up a static website with AWS. There are many [examples on our GitHub repository](https://github.com/pulumi/), but we can clone just the [AWS Python Static Website example](https://github.com/pulumi/examples/tree/master/aws-py-static-website) using a sparse clone:
+
+```bash
+$ mkdir aws-website
+$ cd aws-website
+
+$ git init
+$ git remote add origin -f https://github.com/pulumi/examples/
+$ git config core.sparseCheckout true
+$ echo aws-py-static-website >> .git/info/sparse-checkout
+$ git pull origin master
+```
+
+## Configuring and Deploying the Website
+
+Now that we have a project to build the website, we will need to configure it so that it uses your domain name. Here are the steps to get it running.
+
+1. Create a new stack or instance for testing:
+
+    ```bash
+    $ pulumi stack init website-testing
+    ```
+
+2. Set the AWS region:
+
+    ```bash
+    $ pulumi config set aws:region us-east-1
+    ```
+
+3. Create a Python virtualenv, activate it, and install dependencies:
+
+    This installs the dependent packages [needed](https://www.pulumi.com/docs/intro/concepts/how-pulumi-works/) for our Pulumi program.
+
+    ```bash
+    $ python3 -m venv venv
+    $ source venv/bin/activate
+    $ pip3 install -r requirements.txt
+    ```
+
+4. Edit `Pulumi.website-testing.yaml` to set the configuration parameters. The first parameter is the `targetDomain`, which is the domain for the website (e.g., www.example.com).  The parent domain must be a [Route53 Hosted Zone](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/migrate-dns-domain-in-use.html) in the same AWS account as where the Pulumi program is running. The second parameter is `pathToWebsiteContents`, the relative path to the website’s contents created in Hugo.
+
+    ```yaml
+    config:
+    aws:region: us-east-1
+    static-website:targetDomain: www.<your_domain>.com
+    static-website:pathToWebsiteContents: public
+    ```
```

Comment:
Personally, I'd encourage folks to use `pulumi config set` rather than manually editing the stack's yaml file.

As it's written here, I don't think the yaml content will even work correctly since the stack is named `website-testing` and these config values are prefixed with `static-website` (they'd need to be prefixed with the stack name, `website-testing`).

```suggestion
    ```bash
    $ pulumi config set targetDomain www.<your_domain>.com
    $ pulumi config set pathToWebsiteContents public
    ```
```